sympy/matrices/expressions/tests/test_matrix_exprs.py

```diff
@@ -213,3 +213,32 @@ def test_MatrixElement_doit():
     u = MatrixSymbol('u', 2, 1)
     v = ImmutableMatrix([3, 5])
     assert u[0, 0].subs(u, v).doit() == v[0, 0]
+
+
+def test_identity_powers():
+    M = Identity(n)
+    assert MatPow(M, 3).doit() == M**3
```

Comment:
File deals with the test cases related to the operator **, so I shall change MatPow(M, 3).doit() -> M.
I'll make a few changes so that all sorts of test cases are examined.